### PR TITLE
[MRG+1] Implement doc base image

### PR DESCRIPTION
**Pull request by aaronreidsmith**
# Description

This PR implements the doc base docker image added in alkaline-ml/pmdarima-docker#18. ~It leaves the `documentation` and `doc-requirements` `make` targets, in case someone wants to build the docs locally, but those can easily be deprecated in favor of always using docker to build docs~ I ended up deprecating local building in 245db2b

## Type of change

- [X] Documentation change
- [X] CI/CD Change

# How Has This Been Tested?

- [X] Circle tests pass as expected using the new image

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by aaronreidsmith** [comment]
a7480d1 removes the ability to build docs without docker entirely. This is to make sure our docs are always getting built in the same environment (docker), and to avoid those annoying security emails.

With that being said, a7480d1 fails due to the the `to_dict` test that has been failing with `scipy 1.5.0`, so I probably have to put in a fix for that before I merge this in

**Comment by aaronreidsmith** [comment]
Rebased after #349, so this should be working now

File: Makefile [hunk 1]
```diff
@@ -27,10 +29,11 @@ deploy-twine-test: bdist_wheel deploy-requirements
 		--username ${TWINE_USERNAME} \
 		--password ${TWINE_PASSWORD}
 
-doc-requirements:
-	$(PYTHON) -m pip install -r build_tools/doc/doc_requirements.txt
+documentation: version
+	$(DOCKER) run -v $(HERE):/pmdarima --rm alkalineml/pmdarima-doc-base:latest /bin/bash -c "cd pmdarima && make install && make docker-documentation"
```
Comment (tgsmith61591) [inline Makefile:33]: If you set `-w /pmdarima` you can simplify the entrypoint to `/bin/bash -c "make install && make docker-documentation"`

Comment (aaronreidsmith) [reply Makefile]: Nice! Implemented in 4a9f027

**Pull request merged**